src/util/errorHelpers.ts

```diff
@@ -0,0 +1,55 @@
+export function expectArity(args: IArguments, minArity: number): void {
+	// According to WebIDL overload resolution semantics, only a lower bound applies to the number of arguments provided
+	if (args.length < minArity) {
+		throw new TypeError(`Function should be called with at least ${minArity} arguments`);
+	}
+}
+
+export function expectObject<T>(value: T, Constructor: any): void {
+	if (!(value instanceof Constructor)) {
+		throw new TypeError(`Value should be an instance of ${Constructor.name}`);
```

Comment:
There's one at https://github.com/bwrrp/slimdom.js/blob/align-with-spec/test/Document.tests.ts#L240, although it doesn't test the message.